### PR TITLE
Precedences + PrecedencesTest and puzzle

### PR DESCRIPTION
**Pull request by amihaiemil**
PR for #185 
Added class Precedences and left puzzle for continuing with the stakeholder.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1107](https://github.com/zerocracy/farm/issues/1107) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1107](https://github.com/zerocracy/farm/issues/1107) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1581419627); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by amihaiemil** [comment]
@carlosmiranda I fixed the tests and checkstyle issues here, it's good for review now

File: src/main/java/com/zerocracy/pm/time/Precedences.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                    .set(type)
+                    .up()
+                    .add("predecessor")
+                    .attr("type", prectype)
+                    .set(predecessor)
+                    .up()
+                    .add("successor")
+                    .attr("type", suctype)
+                    .set(successor)
+                    .up()
```
Comment (carlosmiranda) [inline src/main/java/com/zerocracy/pm/time/Precedences.java]: @amihaiemil we don't really need the extra `up()` calls, we're not returning the directives

Comment (carlosmiranda) [reply src/main/java/com/zerocracy/pm/time/Precedences.java]: @amihaiemil we can remove this last one too

**Comment by amihaiemil** [comment]
@carlosmiranda done, I removed the last ``up()`` to :D

File: src/main/java/com/zerocracy/pm/time/Precedences.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+import com.jcabi.xml.XML;
+import com.zerocracy.Item;
+import com.zerocracy.Project;
+import com.zerocracy.Xocument;
+import java.io.IOException;
+import org.xembly.Directives;
+
+/**
+ * A Project's precedences. There are 4 types of precedence: finish-to-start,
+ * start-to-start, finish-to-finish, and start-to-finish. Mode about it, here:
```
Comment (g4s8) [inline src/main/java/com/zerocracy/pm/time/Precedences.java]: @amihaiemil looks like a typo here: `mode` -> `more`? Also please wrap this link with `<a>` tag to make it clickable in javadoc.

**Comment by amihaiemil** [comment]
@g4s8 done, pls have a look :D

**Comment by amihaiemil** [comment]
@g4s8 ping

**Comment by 0crat** [comment]
@carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by amihaiemil** [comment]
@g4s8 ping. Don't forget about this one :)

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1107](https://github.com/zerocracy/farm/issues/1107) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@carlosmiranda According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only one issue was found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by ypshenychka** [comment]
@amihaiemil According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>All commit messages have links to issues

Several commits don't contain link to issue. Please confirm you'll try to avoid this mistake next time.

**Comment by amihaiemil** [comment]
@ypshenychka my mistake; confirmed :D

**Comment by ypshenychka** [comment]
@amihaiemil Thanks!

**Comment by ypshenychka** [comment]
@carlosmiranda ping

**Comment by carlosmiranda** [comment]
@ypshenychka confirmed

**Comment by ypshenychka** [comment]
@carlosmiranda Thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**